### PR TITLE
Votw Legion Dungeon First boss

### PR DESCRIPTION
**Pull request by searcas**
**Changes proposed:**

-  
-  
-  

**Target branch(es):** master

- [x] master

**Issues addressed:**

Closes #  (insert issue tracker number)

**Tests performed:**
[Hatred] Doesn't work at all.
[Darkstrike] Need to attack with both weapons.
[Swoop] Just jumping to target atm, need to damage everyone in his path and knockback all players
[Swoop] Speed of swoop/jump need to be increased in core.
(Does it build, tested in-game, etc.)

**Known issues and TODO list:** (add/remove lines as needed)

- [ Need more scripts on spells] 
- [ Need polishing timers]

### CODE REVIEW

File: dep/recastnavigation/Recast/Source/Recast.cpp [hunk 1]
```diff
@@ -207,7 +207,7 @@ void rcCalcBounds(const float* verts, int nv, float* bmin, float* bmax)
 	rcVcopy(bmax, verts);
 	for (int i = 1; i < nv; ++i)
 	{
-		const float* v = &verts[i*3];
+        const float* v = &verts[i * 3];
```
Comment (Faq) [inline dep/recastnavigation/Recast/Source/Recast.cpp:210]: Usually we don't apply TC code style to 3rd party code.

Comment (illfated) [reply dep/recastnavigation/Recast/Source/Recast.cpp:210]: Thanks, I was a bit uncertain myself.

**Comment by illfated** [comment]
There is an unneeded blank line (Line 15) splitting the last GPL sentence in all of the 100% new .cpp files.
I hope this is all you need to get that fixed.

File: src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp [hunk 2]
```diff
@@ -0,0 +1,554 @@
...
+    DATA_JUMP = 1
+};
+
+enum Actions
+{
+    ACTION_METAMORPHOSIS = 1,
+    ACTION_SUMMON_FEL_FURY = 2,
+};
+const Position jumpPos = { 4284.973f, -451.2752f, 259.66946f, 3.123f };
+const Position spawnFury[4] = {
```
Comment (MaxtorCoder) [inline src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp:155]: I would honestly move these const position lines to the instance_*.h file, looks cleaner imo.

File: src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp [hunk 3]
```diff
@@ -0,0 +1,554 @@
...
+    { 4284.3716f, -441.7253f , 259.60638f},
+};
+//After boss jump he need to jump on his position
+//So some set/get things, didn't figure out
+struct npc_drelanim_whisperwind : public ScriptedAI
+{
+    npc_drelanim_whisperwind(Creature* creature) : ScriptedAI(creature)
+    {
+
+    }
```
Comment (MaxtorCoder) [inline src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp:168]: New line

File: src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp [hunk 4]
```diff
@@ -0,0 +1,554 @@
...
+    {
+        UnitAI::SetData(type, data);
+        if (type == DATA_JUMP && data == 0)
+        {
+            me->GetMotionMaster()->MoveJump(jumpLoc, 20, 0);
+            me->CastSpell(nullptr, SPELL_DRELANIM_WHISPERWIND_KNEEL);
+        }
+    }
+    const Position jumpLoc = { 4229.3887f,-449.48373f, 266.80066f, 3.123f };
+};
```
Comment (MaxtorCoder) [inline src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp:179]: New line as well

File: src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp [hunk 5]
```diff
@@ -0,0 +1,554 @@
...
+    void SetData(uint32 type, uint32 data) override
+    {
+        UnitAI::SetData(type, data);
+        if (type == DATA_JUMP && data == 0)
+        {
+            me->GetMotionMaster()->MoveJump(jumpLoc, 20, 0);
+            me->CastSpell(nullptr, SPELL_DRELANIM_WHISPERWIND_KNEEL);
+        }
+    }
+    const Position jumpLoc = { 4229.3887f,-449.48373f, 266.80066f, 3.123f };
```
Comment (MaxtorCoder) [inline src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp:178]: Same tip as above.

File: src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp [hunk 6]
```diff
@@ -0,0 +1,554 @@
...
+        OnHit += SpellHitFn(spell_dark_strikes::OnHitCheck);
+    }
+    
+};
+
+// ID - 191765 Swoop
+class spell_swoop: public SpellScript
+{
+     PrepareSpellScript(spell_swoop);
+
```
Comment (MaxtorCoder) [inline src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp:414]: Remove extra new line

File: src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp [hunk 7]
```diff
@@ -0,0 +1,554 @@
...
+
+        void HandleCast(SpellEffIndex /*effIndex*/)
```
Comment (MaxtorCoder) [inline src/server/scripts/BrokenIsles/VaultOfTheWardens/boss_tirathon_saltheril.cpp:416]: What the hell is this indentation 😂

File: src/server/scripts/BrokenIsles/VaultOfTheWardens/instance_vault_of_the_wardens.cpp [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+            InstanceScript::OnCreatureCreate(creature);
+            if (instance->IsHeroic())
+            {
+                creature->SetHealth(creature->GetHealth() * 1.33f);
+            }
+            else
+            {
+                creature->SetHealth(creature->GetHealth() * 2.0f);
+            }
+
```
Comment (MaxtorCoder) [inline src/server/scripts/BrokenIsles/VaultOfTheWardens/instance_vault_of_the_wardens.cpp:55]: Fix the lines above, currently normal difficulty will have unit health * 2.0f as well.

**Review by MaxtorCoder — CHANGES_REQUESTED** [review]
There are still a lot of code style nit picks that have to be fixed + maybe change commit name to "2 bosses" instead of "1st boss"

File: src/server/game/Entities/Creature/CreatureData.h [hunk 9]
```diff
@@ -268,7 +268,7 @@ enum CreatureFlagsExtra : uint32
     CREATURE_FLAG_EXTRA_NO_TAUNT             = 0x00000100,       // creature is immune to taunt auras and effect attack me
     CREATURE_FLAG_EXTRA_NO_MOVE_FLAGS_UPDATE = 0x00000200,       // creature won't update movement flags
     CREATURE_FLAG_EXTRA_GHOST_VISIBILITY     = 0x00000400,       // creature will be only visible for dead players
-    CREATURE_FLAG_EXTRA_UNUSED_11            = 0x00000800,
+    CREATURE_FLAG_EXTRA_USE_OFFHAND_ATTACK   = 0x00000800,       // creature will use offhand attacks
```
[the remark this answers is not in the document]
Comment (illfated) [reply src/server/game/Entities/Creature/CreatureData.h:271]: Maybe added in preparation for additional boss scripts?

File: src/server/scripts/BrokenIsles/VaultOfTheWardens/instance_vault_of_the_wardens.cpp [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+        instance_vault_of_the_wardens_InstanceMapScript(InstanceMap* map) : InstanceScript(map) { }
+
+        void Initialize() override
+        {
+            SetBossNumber(DATA_MAX_ENCOUNTERS); 
+        }
+        void OnCreatureCreate(Creature* creature) override
+        {
+            InstanceScript::OnCreatureCreate(creature);
+            if (instance->IsHeroic())
```
Comment (Shauren) [inline src/server/scripts/BrokenIsles/VaultOfTheWardens/instance_vault_of_the_wardens.cpp:47]: Do not touch creature max health in a script, if its broken, let it be broken - this needs to be fixed in `game`, not each individual script

**Pull request closed without merging**